### PR TITLE
Added cubemap-texture type + switched to DOMPointReadOnly

### PR DESCRIPTION
**Pull request by cabanier**
This PR introduces `cubemap-texture`.
More work is needed to clarify how cubemaps work.

### CODE REVIEW

File: webxrlayers-1.bs [hunk 1]
```diff
@@ -446,13 +447,14 @@ The {{XRTextureType}} enum defines what type of texture is allocated.
 <pre class="idl">
 enum XRTextureType {
   "texture",
-  "texture-array"
+  "texture-array",
+  "cubemap-texture"
```
Comment (Artyom17) [inline webxrlayers-1.bs:451]: I am not sure we need publicly expose "cubemap-texture". The cube layer may use only that, but we do not specify it the creation parameters.

Comment (cabanier) [reply webxrlayers-1.bs:451]: You're right. I mistakenly thought it was an attribute. 
I will fix.

File: webxrlayers-1.bs [hunk 2]
```diff
@@ -797,7 +807,7 @@ When this method is invoked, the user agent MUST run the following steps:
   1. Initialize |layer|'s [=XRCompositionLayer/session=] to |session|.
   1. Let |layer|'s {{XRCubeLayer/space}} be the {{session}}'s [=viewer reference space=].
   1. Let |layer|'s {{XRCubeLayer/orientation}} be a new {{DOMPoint}}.
-  1. let |textureType| be {{XRTextureType/"texture-array"}}.
+  1. let |textureType| be {{XRTextureType/"cubemap-texture"}}.
```
Comment (Artyom17) [inline webxrlayers-1.bs:811]: We do not have textureType exposed to user, right?

Comment (toji) [reply webxrlayers-1.bs:811]: No, I just double checked that myself. We shouldn't have this as a publically accessible value, given that it's never actually going to be exposed by the API and it's use is implied by the layer type. The DOMPointReadOnly change is good, though.

File: webxrlayers-1.bs [hunk 3]
```diff
@@ -796,18 +797,27 @@ When this method is invoked, the user agent MUST run the following steps:
...
   1. Initialize |layer|'s [=XRCompositionLayer/session=] to |session|.
   1. Let |layer|'s {{XRCubeLayer/space}} be the {{session}}'s [=viewer reference space=].
-  1. Let |layer|'s {{XRCubeLayer/orientation}} be a new {{DOMPoint}}.
-  1. let |textureType| be {{XRTextureType/"texture-array"}}.
+  1. Let |layer|'s {{XRCubeLayer/orientation}} be a new {{DOMPointReadOnly}}.
   1. let |layout| be |init|'s {{XRLayerInit/layout}}.
   1. If |layout| is {{XRLayerLayout/"stereo-left-right"}} or {{XRLayerLayout/"stereo-top-bottom"}}, throw {{InvalidStateError}} and abort these steps.
-  1. let |layer|'s [=colorTextures=] be the result of [=allocate color textures|allocating color textures=] with |layout|, |textureType|, |context| and |init|.
-  1. let |layer|'s [=depthStencilTextures=] be the result of [=allocate depth textures|allocating depth textures=] with |layout|, |textureType|, |context| and |init|.
+  1. Initialize |array| as follows:
```
Comment (Manishearth) [inline webxrlayers-1.bs:803]:
```suggestion
  1. Initialize |array| as follows, based on the value of |layout|:
```

File: webxrlayers-1.bs [hunk 4]
```diff
@@ -611,17 +611,18 @@ To <dfn>allocate color textures</dfn> using a {{XRLayerInit/layout}} |layout|, a
   1. If |layout| is {{XRLayerLayout/mono}}:
         <dl class="switch">
         <dt> If |textureType| is {{"texture-array"}}:
-        <dd> Initialize |array| with 1 new instance of {{WebGLTexture}} created as a {{TEXTURE_2D_ARRAY}} texture with 1 internal texture using |context| and |init|'s {{XRLayerInit/alpha}}, {{XRLayerInit/pixelWidth}} and {{XRLayerInit/pixelHeight}} values.
+          <dd> Initialize |array| with 1 new instance of {{WebGLTexture}} created as a {{TEXTURE_2D_ARRAY}} texture with 1 internal texture using |context| and |init|'s {{XRLayerInit/alpha}}, {{XRLayerInit/pixelWidth}} and {{XRLayerInit/pixelHeight}} values.
         <dt> Otherwise
-        <dd> Initialize |array| with 1 new instance of {{WebGLTexture}} created as a {{TEXTURE_2D}} texture with |context| and |init|'s {{XRLayerInit/alpha}}, {{XRLayerInit/pixelWidth}} and {{XRLayerInit/pixelHeight}} values.
+          <dd> Initialize |array| with 1 new instance of {{WebGLTexture}} created as a {{TEXTURE_2D}} texture with |context| and |init|'s {{XRLayerInit/alpha}}, {{XRLayerInit/pixelWidth}} and {{XRLayerInit/pixelHeight}} values.
+        <dl>
```
Comment (Manishearth) [inline webxrlayers-1.bs:617]: is there a reason why the cube map case isn't handled differently here? just curious

Comment (cabanier) [reply webxrlayers-1.bs:617]: It turned out pretty messy.
Also, cubemap will require different calculations for width and height which will make it more verbose.

**Pull request merged**